code/mob.dm

```diff
@@ -3130,3 +3135,13 @@
 				. = get_singleton(/datum/pronouns/itIts)
 			else
 				. = get_singleton(/datum/pronouns/theyThem)
+
+/// absorb radiation dose in Sieverts (note 0.4Sv is enough to make someone sick. 2Sv is enough to make someone dead.)
+/mob/proc/take_radiation_dose(Sv)
+	var/rad_res = GET_ATOM_PROPERTY(src,PROP_MOB_RADPROT) || 0 //atom prop can return null, we need it to default to 0
+	if(Sv > 0)
+		src.radiation_dose += (1-(rad_res/(rad_res+1)))*Sv
```

Comment:
weird scaling on the rad protection there, especially since all existing sources are assuming a 0-100(+) percentage scale.